### ISSUE TITLE
BatchUpdate does not consider conversions for default values

### ISSUE THREAD

**Issue by fdahlen** [borisdj/EFCore.BulkExtensions#658]
How to reproduce:
* Configure a DateTimeOffset property called X on your entity to use a Conversion (e.g. DateTimeOffsetToBinaryConverter)
* Add an instance of your entity with a value '2021-11-11' for X, save the db context
* Prepare another instance of your entity, don't set X explicitly (will remain at DateTimeOffset.MinValue)
* Call BatchUpdateAsync, the overload with argument "object updateValues", and pass in your instance without any values set

Current behavior:
* The existing entity in the db context will get its property X overwritten to '0001-01-01'.

Expected behavior:
* The existing entity in the db context will keep its previous X value of  '2021-11-11'.

**Comment by fdahlen** [comment]
I have a potential fix but would like help to see any broader aspects of this. Will add PR.

**Comment by fdahlen** [comment]
See unit test for code sample of how to reproduce.

**Comment by borisdj** [comment]
Trying to integrate your example with existing tests but now not sure if I understand the issue.
Here is changed example in BatchTest:

First on `Item` added property:
`public DateTimeOffset DateTimeOff { get; set; }`

then configured conversion and also defultValue assuming that from title of issue.
```C#
modelBuilder.Entity<Item>().Property(m => m.DateTimeOff)
  .HasConversion(new DateTimeOffsetToBinaryConverter())
  .HasDefaultValue(new DateTimeOffset(new DateTime(2000, 1,1)));
```

And Test being:
```C#
ContextUtil.DbServer = dbServer;
RunDeleteAll(dbServer);

using var context = new TestContext(ContextUtil.GetOptions());

var entity = new Item { Name = "name A", DateTimeOff = new DateTime(2021, 11, 11) };
context.Items.Add(entity);
context.SaveChanges();

//DateTimeOff value in SqlServer DB: 1306054361088000060

var updateTo = new Item();
context.Items.BatchUpdate(updateTo, new List<string> { nameof(Item.DateTimeOff) });

//DateTimeOff value in SqlServer DB now being: 0
```
May first assumption was that you expect value not to be zero 0 but to have number corresponding to 2000,1,1.
But this remains zero, even with your fix.
Also you can get that value easily by just setting it explicitly: `var updateTo = new Item { DateTimeOff = new DateTime(2000, 1, 1) };`
So can you elaborate more and update this example to your situation.

**Comment by fdahlen** [comment]
I'm sorry if the steps to reproduce were unclear. We are not using any .HasDefaultValue() on the model, nor update columns. The issue is that when updating _other properties_ on the item, those properties on the item that _have_ conversions will get updated as well, even if they are not mentioned in the update columns. The values gets set to the default value of the property type. Basically overwriting my item's other data.

The merged PR contains a unit test that demonstrates the issue, which I hope clarifies the actual problem. That unit test fails without the code change.

**Comment by borisdj** [comment]
Got it, thx for contrib.
Test updated, and v6.0.2 published.
I've added fix also to Branch 5 and publish v5.4.1, closed that PR because already committed same change.

**Comment by robjackstewart** [comment]
I am experiencing this issue and am tied to .NET Core 3.1, could the same fix be applied to the 3.x version of the package?

**Comment by borisdj** [comment]
v3 is not actively maintained.
Best if you could upgrade soon to EF 5 or 6.
Otherwise if you can add this fix to v3 branch and make a PR a could published new Nuget.

**Comment by robjackstewart** [comment]
@borisdj Ive taken @fdahlen's fix and applied it to the EFCore3.1 branch but I am  unable to push my branch to the repo. Do you need to grant rights to push a branch?

**Comment by fdahlen** [comment]
@robjackstewart You need to fork and submit a PR from there towards this repo. See the following guide:

https://docs.github.com/en/get-started/quickstart/contributing-to-projects

**Comment by robjackstewart** [comment]
Thanks @fdahlen! 

Please find [PR for EFCore 3.1 here](https://github.com/borisdj/EFCore.BulkExtensions/pull/764).

**Comment by fdahlen** [comment]
@robjackstewart No problem, happy to see my fix can help someone else. 

I’m no maintainer though so you’ll have to wait for @borisdj.

**Comment by borisdj** [comment]
PR Merged and v3.6.6 published.
